pkg/controller/container-runtime-config/helpers_test.go

```diff
@@ -1245,22 +1263,148 @@ func TestGetValidScopePolicies(t *testing.T) {
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "a.com", Mirrors: []apicfgv1.ImageMirror{"a.mirror"}},
+							{Source: "a.com/a1", Mirrors: []apicfgv1.ImageMirror{"a-a1.mirror/a1"}},
+							{Source: "x.com/x1/x2", Mirrors: []apicfgv1.ImageMirror{"x-x1-x2.mirror/x1/x2"}},
+						},
+					},
+				},
+			},
+			expectedSigstoreRegistriesConfigScopes: []string{"*.example.com", "a-a1.mirror/a1/a2", "a.com/a1/a2", "a.com/a1/a2@sha256:0000000000000000000000000000000000000000000000000000000000000000", "policy.scope"},
+		},
+		{
+			name:                 "cip wildcard scope *.example.com is the mirror source",
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "*.example.com", Mirrors: []apicfgv1.ImageMirror{"matched.mirror"}},
+						},
+					},
+				},
+			},
+			expectedSigstoreRegistriesConfigScopes: []string{"*.example.com", "a.com/a1/a2", "a.com/a1/a2@sha256:0000000000000000000000000000000000000000000000000000000000000000", "policy.scope", "matched.mirror"},
+		},
+		{
+			name:                 "cip scope *.example.com and mirror source have wildcard matching",
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "a.example.com", Mirrors: []apicfgv1.ImageMirror{"a-example.mirror"}},
+							{Source: "*.x.example.com", Mirrors: []apicfgv1.ImageMirror{"matched.mirror"}},
+							{Source: "*.scope", Mirrors: []apicfgv1.ImageMirror{"start-scope.mirror"}},
+							{Source: "x/x1/x2", Mirrors: []apicfgv1.ImageMirror{"x-x1-x2.mirror/x1/x2"}},
+						},
+					},
+				},
+			},
+			expectedSigstoreRegistriesConfigScopes: []string{"*.example.com", "a-example.mirror", "a.com/a1/a2", "a.com/a1/a2@sha256:0000000000000000000000000000000000000000000000000000000000000000", "a.example.com", "matched.mirror", "policy.scope", "start-scope.mirror"},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			registriesTOML, err := updateRegistriesConfig(templateRegistriesConfig, nil, nil, tc.icspRules, tc.idmsRules, tc.itmsRules)
+			require.NoError(t, err)
+			got, err := generateSigstoreRegistriesdConfig(tc.clusterScopePolicies, registriesTOML)
+			require.NoError(t, err)
+
+			gotRegistriesCfg := &registriesConfig{}
+			err = yaml.Unmarshal(got, &gotRegistriesCfg)
+			require.NoError(t, err)
+
+			require.Len(t, gotRegistriesCfg.Docker, len(tc.expectedSigstoreRegistriesConfigScopes))
+			for _, scope := range tc.expectedSigstoreRegistriesConfigScopes {
```

Comment:
`assert.ElementsMatch(…, expected, maps.Keys(gotRegistriesCfg.Docker))` would check not just a subset relationship, but equality.